In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("listings.csv")

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 33329 entries, 0 to 33328
Data columns (total 74 columns):
 #   Column                                        Non-Null Count  Dtype  
---  ------                                        --------------  -----  
 0   id                                            33329 non-null  int64  
 1   listing_url                                   33329 non-null  object 
 2   scrape_id                                     33329 non-null  int64  
 3   last_scraped                                  33329 non-null  object 
 4   name                                          33327 non-null  object 
 5   description                                   32231 non-null  object 
 6   neighborhood_overview                         20108 non-null  object 
 7   picture_url                                   33329 non-null  object 
 8   host_id                                       33329 non-null  int64  
 9   host_url                                      33329 non-null 

# Task 1

In [4]:
df.neighbourhood_cleansed.value_counts() 

# How many listings are there in "neighbourhood_cleansed"

Hollywood               1580
Venice                  1567
Long Beach              1304
Santa Monica            1099
Downtown                 977
                        ... 
Chatsworth Reservoir       1
Elizabeth Lake             1
Leona Valley               1
South Diamond Bar          1
Bradbury                   1
Name: neighbourhood_cleansed, Length: 266, dtype: int64

In [5]:
df[df.neighbourhood_cleansed.str.contains("University Park")].neighbourhood_cleansed.value_counts()

#  Filter listings to "University Park"

University Park    76
Name: neighbourhood_cleansed, dtype: int64

In [6]:
university_park = df[df.neighbourhood_cleansed == "University Park"].copy()

In [7]:
len(university_park)

76

There are 76 listings in Downtown

In [8]:
university_park.to_csv("university_park.csv")

# Create a new dataframe named "university_park.csv"

# Task 2

In [9]:
df.price

0        $115.00
1         $89.00
2        $167.00
3         $50.00
4        $118.00
          ...   
33324    $155.00
33325    $155.00
33326    $155.00
33327    $152.00
33328    $143.00
Name: price, Length: 33329, dtype: object

How to clean the string type in to float type

In [10]:
def cleanup(x):
    y = x.split("$")
    z = y[1].replace(',','')
    z = float(z)
    
    return z

# function to clean a string type cloumn into float type

In [11]:
p =  "$1,150.00"

In [12]:
p.split("$")[1]

'1,150.00'

In [13]:
p.split("$")[1].replace(",","")

'1150.00'

In [14]:
df["price"]

0        $115.00
1         $89.00
2        $167.00
3         $50.00
4        $118.00
          ...   
33324    $155.00
33325    $155.00
33326    $155.00
33327    $152.00
33328    $143.00
Name: price, Length: 33329, dtype: object

In [15]:
df_price = df.price.apply(cleanup)

In [16]:
df_price

0        115.0
1         89.0
2        167.0
3         50.0
4        118.0
         ...  
33324    155.0
33325    155.0
33326    155.0
33327    152.0
33328    143.0
Name: price, Length: 33329, dtype: float64

In [17]:
df_price.nunique()

1330

There are 1330 different variations are under the "price" column

In [18]:
# Another method
"""
    function : turns string to float in X
    param X : input column with strings inside
    return X_float : return column with float
"""

"""
def cleanup(x):

    for ele in x :
        
        ele = ele[1:].split(",")
        
        if(len(ele) == 2):
            ele = ele[0] + ele[1]
        else:
            ele = ''.join(ele)
        
        print(ele)
        
    return x
    
df.price_cleanup = cleanup(df.price)

df.price_cleanup.nunique()

"""

'\ndef cleanup(x):\n\n    for ele in x :\n        \n        ele = ele[1:].split(",")\n        \n        if(len(ele) == 2):\n            ele = ele[0] + ele[1]\n        else:\n            ele = \'\'.join(ele)\n        \n        print(ele)\n        \n    return x\n    \ndf.price_cleanup = cleanup(df.price)\n\ndf.price_cleanup.nunique()\n\n'

# Task 3

In [19]:
university_park.price

961       $50.00
2283      $20.00
2592      $50.00
2764      $35.00
2781     $100.00
          ...   
26477     $35.00
26603    $250.00
27833     $40.00
29372     $94.00
32774     $90.00
Name: price, Length: 76, dtype: object

How to find the average price in the University Park

In [20]:
university_park["clean_price"] = university_park.price.apply(cleanup)
# create a new column called "clean_price"

In [21]:
university_park.clean_price

961       50.0
2283      20.0
2592      50.0
2764      35.0
2781     100.0
         ...  
26477     35.0
26603    250.0
27833     40.0
29372     94.0
32774     90.0
Name: clean_price, Length: 76, dtype: float64

In [22]:
university_park.clean_price.mean()

72.97368421052632

In [23]:
round(university_park.clean_price.mean(),2)

72.97

The average price of the University Park is $72.97

# Task 4

In [24]:
university_park.sort_values("id", ascending= True).head()

# sort by "id" column

id                           listing_url       scrape_id  \
961   1630098  https://www.airbnb.com/rooms/1630098  20211205213202   
2283  4708239  https://www.airbnb.com/rooms/4708239  20211205213202   
2592  5627628  https://www.airbnb.com/rooms/5627628  20211205213202   
2764  6092506  https://www.airbnb.com/rooms/6092506  20211205213202   
2781  6133492  https://www.airbnb.com/rooms/6133492  20211205213202   

     last_scraped                                              name  \
961    2021-12-07  USC 1 block north -Private Room/Private Bath B&B   
2283   2021-12-06                  Furnished room minutes from DTLA   
2592   2021-12-06                          Nupac Apartment near USC   
2764   2021-12-06               USC ICON Plaza Summer Sublease - Pr   
2781   2021-12-07     [TESTING] Spacious Private Room near DTLA/USC   

                                            description  \
961   Private room and private bath Bed and Breakfas...   
2283  Fully furnished private room in a 3BHK for upt...   
2592  The apartment is 10 min walking distance from ...   
2764  Looking for a FEMALE to take over my private, ...   
2781  Note: not available right now, just testing th...   

                                  neighborhood_overview  \
961   1 block from USC in historic West Adams distri...   
2283                                                NaN   
2592                                                NaN   
2764                                                NaN   
2781                                                NaN   

                                            picture_url   host_id  \
961   https://a0.muscache.com/pictures/f3393619-30e0...   8663618   
2283  https://a0.muscache.com/pictures/59117140/05d9...  15973150   
2592  https://a0.muscache.com/pictures/74195073/fcd8...  21364327   
2764  https://a0.muscache.com/pictures/75932141/a820...  31609605   
2781  https://a0.muscache.com/pictures/109592346/4b7...  17090189   

                                        host_url  ... review_scores_location  \
961    https://www.airbnb.com/users/show/8663618  ...                   4.67   
2283  https://www.airbnb.com/users/show/15973150  ...                   2.50   
2592  https://www.airbnb.com/users/show/21364327  ...                    NaN   
2764  https://www.airbnb.com/users/show/31609605  ...                    NaN   
2781  https://www.airbnb.com/users/show/17090189  ...                    NaN   

     review_scores_value                                       license  \
961                 4.63  Exempt - This listing is a bed and breakfast   
2283                2.50                                           NaN   
2592                 NaN                                           NaN   
2764                 NaN                                           NaN   
2781                 NaN                                           NaN   

     instant_bookable calculated_host_listings_count  \
961                 f                              2   
2283                f                              1   
2592                f                              1   
2764                f                              1   
2781                f                              1   

     calculated_host_listings_count_entire_homes  \
961                                            0   
2283                                           0   
2592                                           0   
2764                                           0   
2781                                           0   

     calculated_host_listings_count_private_rooms  \
961                                             2   
2283                                            1   
2592                                            0   
2764                                            1   
2781                                            1   

     calculated_host_listings_count_shared_rooms reviews_per_month clean_price  
961                                            0        

In [25]:
university_park.sort_values("id", ascending= True).to_csv("university-park.csv", index=False)

# Extra question

In [26]:
df[['price', 'host_url']]

price                                     host_url
0      $115.00        https://www.airbnb.com/users/show/521
1       $89.00       https://www.airbnb.com/users/show/3008
2      $167.00       https://www.airbnb.com/users/show/3041
3       $50.00       https://www.airbnb.com/users/show/3207
4      $118.00       https://www.airbnb.com/users/show/3415
...        ...                                          ...
33324  $155.00   https://www.airbnb.com/users/show/16760875
33325  $155.00   https://www.airbnb.com/users/show/16760875
33326  $155.00   https://www.airbnb.com/users/show/16760875
33327  $152.00  https://www.airbnb.com/users/show/375448432
33328  $143.00  https://www.airbnb.com/users/show/205333524

[33329 rows x 2 columns]

In [27]:
df_extra = df[['price', 'host_url']]

In [28]:
df_extra["clean_price"] = df_extra.price.apply(cleanup)

C:\Users\William Xu\AppData\Local\Temp/ipykernel_52472/1272134181.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_extra["clean_price"] = df_extra.price.apply(cleanup)


In [29]:
df_extra.sort_values('clean_price', ascending=False).head()

price                                     host_url  clean_price
25712  $25,000.00  https://www.airbnb.com/users/show/345121555      25000.0
25686  $25,000.00  https://www.airbnb.com/users/show/345121555      25000.0
25707  $25,000.00  https://www.airbnb.com/users/show/345121555      25000.0
25708  $25,000.00  https://www.airbnb.com/users/show/345121555      25000.0
25710  $25,000.00  https://www.airbnb.com/users/show/345121555      25000.0

In [30]:
"""
# Another method to find the max value of the "price"

df_extra[df_extra['clean_price'] == df_extra['clean_price'].max()].head()

"""

'\n# Another method to find the max value of the "price"\n\ndf_extra[df_extra[\'clean_price\'] == df_extra[\'clean_price\'].max()].head()\n\n'

In [31]:
df_extra.loc[df_extra['clean_price'] == 25000.0]

# Find the host URL with the price in $25,000

price                                     host_url  clean_price
25686  $25,000.00  https://www.airbnb.com/users/show/345121555      25000.0
25707  $25,000.00  https://www.airbnb.com/users/show/345121555      25000.0
25708  $25,000.00  https://www.airbnb.com/users/show/345121555      25000.0
25710  $25,000.00  https://www.airbnb.com/users/show/345121555      25000.0
25711  $25,000.00  https://www.airbnb.com/users/show/345121555      25000.0
25712  $25,000.00  https://www.airbnb.com/users/show/345121555      25000.0

The most expensive price is $25,000, and 6 host URL with this price could be found above